### PR TITLE
LG-3858: Date Picker calendar button overflow

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3858(https://jira.mongodb.org/browse/LG-3858)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/date-picker/src/shared/components/DateInput/DateFormField/DateFormField.styles.ts [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+  [Size.Large]: 28,
+};
+
+// FormField does not account for icon width combined with the gap so this is a temp workaround
+const getMargin = (size: Size) => {
+  const iconWidthDiff = (calendarButtonSize[size] - 16) / 2; // spec is 16px width but eng width is wider because of hover/focus states
+  return -iconWidthDiff;
+};
+
+export const iconButtonStyles = (size: Size) => css`
```
Comment (TheSonOfThomp) [inline packages/date-picker/src/shared/components/DateInput/DateFormField/DateFormField.styles.ts:17]: his could be a static object.
(Though it's debatable whether this is better. _Technically_ this function will be called on every render, but the perf impact is negligible)

```js
iconButtonStyles = {
 [Size.Small]: css`
    svg + button {
      margin-left: ${getMargin(Size.Small)}px;
    }
`
}
```

File: .changeset/yellow-dolls-eat.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/form-field': patch
+---
+
+Removes 8px gap between icons. The gap will always be 4px.
```
Comment (TheSonOfThomp) [inline .changeset/yellow-dolls-eat.md:5]: "Changes the gap between icons from 8px to 4px for all sizes."

**Review by TheSonOfThomp — APPROVED** [review]
Small nits, but LGTM

File: .changeset/yellow-dolls-eat.md (hunk 2 above)
Comment (shaneeza) [reply .changeset/yellow-dolls-eat.md:5]: this is better; mine sound threatening.

**Pull request merged**